pkg/types/vm.go

```diff
@@ -26,3 +25,23 @@ type VM struct {
 
 	Tags []Tag `json:"tags"`
 }
+
+type VMStatus string
+
+const (
+	VMStatusQueued       VMStatus = "queued"       // Not assigned to a runner yet
+	VMStatusAssigned     VMStatus = "assigned"     // Assigned to a runner, but have not heard back from the runner
+	VMStatusPreparing    VMStatus = "preparing"    // The runner sets this when is receives the request
+	VMStatusProvisioning VMStatus = "provisioning" // The runner sets this when it starts provisioning
+	VMStatusRunning      VMStatus = "running"      // The runner sets this when it is done provisioning or upgrading and available
+	VMStatusTerminated   VMStatus = "terminated"   // This is set when the cluster expires or is deleted
+	VMStatusError        VMStatus = "error"        // Something unexpected
+	VMStatusDeleted      VMStatus = "deleted"
+)
+
+type VMVersion struct {
+	Name          string                     `json:"short_name"`
+	Versions      []string                   `json:"versions"`
+	InstanceTypes []string                   `json:"instance_types"`
+	Status        *ClusterDistributionStatus `json:"status,omitempty"`
```

Comment:
Todo refactor `ClusterDistributionStatus` into `DistributionStatus` or make it `VMDistributionStatus`